### PR TITLE
WEB-3357: Nestjs - migrate campaign admin CRUD operations

### PR DESCRIPTION
**Pull request by oldfashionedrobot**
- Adds Admin Module to house admin controllers
- Adds AdminCampaigns controller + service for campaign CRUD
- Updates the way I was using Zod + ZodValidationPipe for campaigns CRUD (cleaner to use classes and not have to manually pass in the schema object to ZodValidationPipe instances)
- Updated the raw SQL query in the `campaignsService` to use Prisma query objects
- Installed `validator` package for premade validators for many cases 
- Also added a stub for the admin-create-email endpoint (for sending a password reset for an admin created campaign user)

Postman tests: https://goodpartyorg.postman.co/workspace/GP-API~66e2b59d-bf6a-4380-81c2-1cbf2d01bddf/run/38094541-b98eb38b-3452-4811-947a-ba757b3ca21b

### CODE REVIEW

File: src/admin/campaigns/adminCampaigns.controller.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { AdminUpdateCampaignSchema } from './schemas/adminUpdateCampaign.schema'
+import { AdminSendCreateEmailSchema } from './schemas/adminSendCreateEmail.schema'
+
+@Controller('admin/campaigns')
+@UsePipes(ZodValidationPipe)
+export class AdminCampaignsController {
+  constructor(private readonly adminCampaignsService: AdminCampaignsService) {}
+
+  @Post()
+  async create(@Body() body: AdminCreateCamapaignSchema) {
```
Comment (RavenHursT) [inline src/admin/campaigns/adminCampaigns.controller.ts:25]: Typo here.

File: prisma/schema/user.prisma [hunk 2]
```diff
@@ -1,3 +1,8 @@
+enum UserRole {
+  sales
+  campaign
```
Comment (tomer-tgp) [inline prisma/schema/user.prisma:3]: admin?

File: src/admin/campaigns/adminCampaigns.service.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      where: {
+        email: {
+          equals: email,
+          mode: 'insensitive',
+        },
+      },
+    })
+
+    if (exists > 0) {
+      return 'Email already in use'
```
Comment (tomer-tgp) [inline src/admin/campaigns/adminCampaigns.service.ts:30]: this should probably return a 409 not a 400

**Review by tomer-tgp — CHANGES_REQUESTED** [review]

File: prisma/schema/user.prisma (hunk 2 above)
Comment (tomer-tgp) [reply prisma/schema/user.prisma:3]: or we can add "isAdmin" to User

**Review by tomer-tgp — APPROVED** [review]

**Pull request merged**